### PR TITLE
Convert MissionPlanner script to MAVLink reader.

### LINKED ISSUE #209: Error with auvsi_mp.py and Mission Planner
Hi, I'm trying to get this year's version of auvsi_mp running in mission planner and I keep running into this, even across different machines: "Error running script expected IList[Byte], got str" I haven't had much luck resolving it, any ideas?

### PR DESCRIPTION
**Pull request by pmtischler**
Add functionality to interop_cli.py to read MAVLink streams and forward
telemetry messages to interoperability server. Recommend using MavProxy
to tee traffic to the CLI. Add e2e tests by replaying log recorded using
Ardupilot simulator.

Fixed #209

Note that I haven't been able to test message conversion validity with a simulator-generated log. Using Google Maps and an altitude lookup, it appears to be for a valid launch location at an Australian flying club. Post-submit, I'll reach out to someone with an autopilot to validate.

### CODE REVIEW

**Comment by pmtischler** [comment]
@DarkmatterVale @imritzror @JackHarrington @s4lvi
FYI

File: client/tools/README.md [hunk 1]
```diff
@@ -4,8 +4,8 @@ AUVSI SUAS: Interoperability Client Tools
 interop_cli.py
 --------------
 
-This command line tool can be used to get mission details, upload targets, and
-probe the server with dummy data.
+This command line tool can be used to get mission details, upload targets,
+probe the server with dummy data, and forwards MAVLink packets to interop.
```
Comment (yogishadixit) [inline client/tools/README.md]: s/forwards/forward/

Comment (pmtischler) [inline client/tools/README.md]: Done.

**Comment by pmtischler** [comment]
@prattmic @yogishadixit If you'd like to further review, please send comments by EOD Monday. Otherwise I'll submit (this PR has been open 8d).

File: client/tools/proxy_mavlink.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    Args:
+        device: A pymavlink device name to forward.
+        client: Interop Client with which to send telemetry packets.
+    """
+    # Create the MAVLink connection.
+    mav = mavutil.mavlink_connection(device, autoreconnect=True)
+
+    # Track rates.
+    print_period = 5.0
```
Comment (yogishadixit) [inline client/tools/proxy_mavlink.py]: Define this as a constant at the top?

File: client/tools/proxy_mavlink.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            logger.critical(
+                'Did not receive MAVLink packet for over 10 seconds.')
+            sys.exit(-1)
+        # Convert to telemetry.
+        telemetry = Telemetry(latitude=mavlink_latlon(msg.lat),
+                              longitude=mavlink_latlon(msg.lon),
+                              altitude_msl=mavlink_alt(msg.alt),
+                              uas_heading=mavlink_heading(msg.hdg))
+        # Forward telemetry.
+        client.post_telemetry(telemetry)
```
Comment (yogishadixit) [inline client/tools/proxy_mavlink.py]: Handle error from interop server?

Comment (pmtischler) [inline client/tools/proxy_mavlink.py]: Done. Note I've maintained exit-on-failure, which was intentional.

File: client/tools/proxy_mavlink.py (hunk 2 above)
Comment (pmtischler) [inline client/tools/proxy_mavlink.py]: Done.

**Pull request merged**